wallet/src/main/scala/org/bitcoins/wallet/internal/FundTransactionHandling.scala

```diff
@@ -71,24 +72,52 @@ trait FundTransactionHandling extends WalletLogger { self: Wallet =>
       fromTagOpt: Option[AddressTag],
       markAsReserved: Boolean): Future[
     FundRawTxHelper[ShufflingNonInteractiveFinalizer]] = {
+    val action = fundRawTransactionInternalAction(destinations,
+                                                  feeRate,
+                                                  fromAccount,
+                                                  coinSelectionAlgo,
+                                                  fromTagOpt,
+                                                  markAsReserved)
+
+    for {
+      (txHelper, callbackF) <- safeDatabase.run(action)
+      _ <- callbackF
+    } yield txHelper
+  }
+
+  private[bitcoins] def fundRawTransactionInternalAction(
+      destinations: Vector[TransactionOutput],
+      feeRate: FeeUnit,
+      fromAccount: AccountDb,
+      coinSelectionAlgo: CoinSelectionAlgo = CoinSelectionAlgo.LeastWaste,
+      fromTagOpt: Option[AddressTag],
+      markAsReserved: Boolean): DBIOAction[
+    (FundRawTxHelper[ShufflingNonInteractiveFinalizer], Future[Unit]),
```

Comment:
Lets just move the `Future[Unit]` into the `FundRawTxHelper`? That way it can have a descriptive named like `reservedUtxoCallbackF` or something like that.

Reply:
done